### PR TITLE
Exclude ids from artist default list

### PR DESCRIPTION
**Pull request by xtina-starr**
closes: https://github.com/artsy/collector-experience/issues/842

In addition, we refactored the `PopularArtists` and `ArtistSearchResults` components to remove duplication and use a generic `ItemLink` component

### CODE REVIEW

File: src/Components/Onboarding/Steps/Artists/ArtistSearchResults.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 
-export interface RelayProps {
+export interface Props {
   term: string
-  viewer: {
+}
+
+interface RelayProps extends React.HTMLProps<HTMLAnchorElement>, Props {
+  relay?: RelayProp
+  viewer?: {
```
Comment (mzikherman) [inline src/Components/Onboarding/Steps/Artists/ArtistSearchResults.tsx]: Is this optional?

**Comment by mzikherman** [comment]
Looks great! Feel free to merge, just the 1 minor comment.

**Comment by alloy** [comment]
Re-assigning to @mzikherman since he’s already done a review.

File: src/Components/Onboarding/Steps/Artists/ArtistSearchResults.tsx (hunk 1 above)
Comment (xtina-starr) [reply src/Components/Onboarding/Steps/Artists/ArtistSearchResults.tsx]: this shouldn't be optional. fixed it.

**Pull request merged**